documentation/quick-start.md

```diff
@@ -0,0 +1,236 @@
+# Quick Start Guide
+
+## Getting and posting data with promises
+
+The simplest `GET` request:
+
+```js
+import got from 'got';
+
+const url = 'https://httpbin.org/anything';
+const response = await got(url);
+```
+
+The call returns a <code>Promise<[Response](3-streams.md#response-1)></code>. If the body contains JSON, it can be retrieved directly:
+
+```js
+import got from 'got';
+
+const url = 'https://httpbin.org/anything';
+const data = await got(url).json();
+```
+
+The similar <code>[promise.text](1-promise.md#promisetext)</code> method returns plain text.
+
+All `got` methods accept an options object for passing extra configuration, such as headers:
+
+```js
+import got from 'got';
+
+const url = 'https://httpbin.org/anything';
+
+const options = {
+	headers: {
+		'Custom-Header': 'Quick start',
+	},
+	timeout: {
+		send: 3500
+	},
+};
+
+const data = await got(url, options).json();
+```
+
+A `POST` request is very similar:
+
+```js
+import got from 'got';
+
+const url = 'https://httpbin.org/anything';
+
+const options = {
+	json: {
+		documentName: 'Quick Start',
+	},
+};
+
+const data = await got.post(url, options);
+```
+
+The request body is passed in the options object. The `json` property will automatically set headers accordingly. Custom headers can be added exactly as above.
+
+## Using streams
+
+The [Stream API](3-streams.md) allows to leverage [Node.js Streams](https://nodejs.dev/learn/nodejs-streams) capabilities:
+
+```js
+import {open} from 'node:fs/promises';
+import {pipeline} from 'node:stream/promises';
+import got from 'got';
+
+const url = 'https://httpbin.org/anything';
+
+const options = {
+	json: {
+		documentName: 'Quick Start',
+	},
+};
+
+const gotStream = got.stream.post(url, options);
+
+const fileDescriptor = await open('anything.json', 'w');
```

Comment:
Does the file descriptor need to be closed after pipeline?

Reply:
You're right it is cleaner. I'll add a finally block to the `try { } catch { }`

Reply:
Do we need to use `fs.open` at all, why can't we just `fs.createWriteStream`?

Reply:
The idea was to uniformly use the new promise API, as for streams. Before commit 6340f64ee92d32b9499cd721e25e9ab8ebc1d990 we used `fs.createWriteStream`. I don't mind, what do you think ? Less code is better ?

Reply:
> The idea was to uniformly use the new promise API, as for streams.

Let's not blindly use the APIs. What do you need the file descriptor for? If it's not needed then don't use it :)